features/netsocket/SocketAddress.cpp

```diff
@@ -122,6 +122,7 @@ static void ipv6_from_address(uint8_t *bytes, const char *addr)
     // Move suffix to end
     memmove(&shorts[NSAPI_IPv6_BYTES/2-suffix], &shorts[0],
             suffix*sizeof(uint16_t));
+    memset(&shorts[0], 0, suffix*sizeof(uint16_t));
```

Comment:
The remaining space has already been zeroed out by the previous memset:
https://github.com/geky/mbed/blob/16c4da03e62fc2367b33be8bb9358c28804fffb3/features/netsocket/SocketAddress.cpp#L110

Either way would work